pkg/meta/model/job_args.go

```diff
@@ -109,3 +110,98 @@ func GetTruncateTableArgsAfterRun(job *Job) (*TruncateTableArgs, error) {
 	}
 	return argsV2, nil
 }
+
+// RenameTableArgs is the arguements for rename table DDL job.
+type RenameTableArgs struct {
+	// for Args
+	OldSchemaID  int64        `json:"old_schema_id,omitempty"`
+	SchemaName   pmodel.CIStr `json:"schema_name,omitempty"`
+	NewTableName pmodel.CIStr `json:"new_table_name,omitempty"`
+
+	// for CtxVars
+	// SchemaIDs contains {oldSchemaID, newSchemaID}
+	SchemaIDs []int64 `json:"-"`
+	// TableIDs contains {tableID}
+	TableIDs []int64 `json:"-"`
+}
+
+func (rt *RenameTableArgs) fillJob(job *Job) {
+	if job.Version == JobVersion1 {
+		job.Args = []any{rt.OldSchemaID, rt.NewTableName, rt.SchemaName}
+		job.CtxVars = []any{rt.SchemaIDs, rt.TableIDs}
+	} else {
+		job.ArgsV2 = rt
+	}
+}
+
+// NewRenameTableArgs creates a struct with given parameters.
+func NewRenameTableArgs(
+	oldSchemaID int64, newSchemaID int64, schemaName pmodel.CIStr,
+	tableID int64, newTableName pmodel.CIStr,
+) *RenameTableArgs {
+	return &RenameTableArgs{
+		OldSchemaID:  oldSchemaID,
+		SchemaName:   schemaName,
+		NewTableName: newTableName,
+		SchemaIDs:    []int64{oldSchemaID, newSchemaID},
+		TableIDs:     []int64{tableID},
+	}
+}
+
+// GetRenameTableArgs get the arguements from job.
+func GetRenameTableArgs(job *Job) (*RenameTableArgs, error) {
+	if job.Version == JobVersion1 {
+		var (
+			oldSchemaID  int64
+			schemaName   pmodel.CIStr
+			newTableName pmodel.CIStr
+		)
+
+		// decode args and cache in args.
+		if len(job.Args) == 0 {
+			err := job.DecodeArgs(&oldSchemaID, &newTableName, &schemaName)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			job.Args = []any{oldSchemaID, newTableName, schemaName}
+		} else {
+			oldSchemaID = job.Args[0].(int64)
+			schemaName = job.Args[1].(pmodel.CIStr)
+			newTableName = job.Args[2].(pmodel.CIStr)
+		}
+
+		args := RenameTableArgs{
+			OldSchemaID:  oldSchemaID,
+			SchemaName:   schemaName,
+			NewTableName: newTableName,
+		}
+		if len(job.CtxVars) > 0 {
+			args.SchemaIDs = job.CtxVars[0].([]int64)
+			args.TableIDs = job.CtxVars[1].([]int64)
+		}
+		return &args, nil
+	}
+
+	argsV2, err := getOrDecodeArgsV2[*RenameTableArgs](job)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return argsV2, err
+}
+
+// GetRenameTableUniqueIDs gets unique IDs from job for rename table type.
+func GetRenameTableUniqueIDs(job *Job, schema bool) []int64 {
+	args := &RenameTableArgs{}
+
+	if job.Version == JobVersion1 {
+		args.SchemaIDs = job.CtxVars[0].([]int64)
```

Comment:
we don't need to use `CtxVars` now even for version 1, just store it in the `args` without persist them

we don't need to write a exported method for a only used-once logic

and i found a better way to pass those args during submitting, see https://github.com/pingcap/tidb/pull/55964/files#diff-7328439fbfe5c8d8a925b3de0a38dd6393f097a8015e448d5f8ec1a0e456fe24R156-R157